rhino/src/main/java/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -1051,7 +1051,20 @@ public static String toString(Object val) {
     static String defaultObjectToString(Scriptable obj) {
         if (obj == null) return "[object Null]";
         if (Undefined.isUndefined(obj)) return "[object Undefined]";
-        return "[object " + obj.getClassName() + ']';
+
+        String tag = null;
+        if (ScriptableObject.hasProperty(obj, SymbolKey.TO_STRING_TAG)) {
+            Object tagValue = ScriptableObject.getProperty(obj, SymbolKey.TO_STRING_TAG);
+            if (tagValue instanceof String) {
```

Comment:
please check for CharSequence here to support at lease ConsString also

Reply:
Thanks for your feedback! Now, check CharSequence